packages/api/src/derivedProperties/WithPropertyObjectSet.ts

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { ValidAggregationKeys } from "../aggregate/AggregatableKeys.js";
+import type { WhereClause } from "../aggregate/WhereClause.js";
+import type {
+  ObjectOrInterfaceDefinition,
+  PropertyKeys,
+} from "../ontology/ObjectOrInterface.js";
+import type {
+  CompileTimeMetadata,
+  PropertyDef,
+} from "../ontology/ObjectTypeDefinition.js";
+import type { LinkedType, LinkNames } from "../util/LinkUtils.js";
+import type { WithPropertyDefinition } from "./WithPropertiesClause.js";
+
+export interface WithPropertyObjectSet<Q extends ObjectOrInterfaceDefinition>
+  extends
+    BaseWithPropertyObjectSet<Q>,
+    AggregatableWithPropertyObjectSet<Q>,
+    SingleLinkWithPropertyObjectSet<Q>
+{}
+
+export interface BaseWithPropertyObjectSet<
```

Comment:
This object set only has filter and pivot as aggregations and selectProperty is not available on the base object set.